modules/nextflow/src/main/groovy/nextflow/scm/AzureRepositoryProvider.groovy

```diff
@@ -39,21 +39,36 @@ final class AzureRepositoryProvider extends RepositoryProvider {
     private String continuationToken
 
     AzureRepositoryProvider(String project, ProviderConfig config=null) {
-        /*
-        Azure repo format follows Organization/Project/Repository where Project can be optional
-        If Project is not present then Repository is used as Project (and also as Repository)
-         */
         this.urlPath = project
-        def tokens = project.tokenize('/')
+        def tokens = getUniformPath(project)
         this.repo = tokens.removeLast()
-        if( tokens.size() == 1){
-            this.project = [tokens.first(), this.repo].join('/')
-        }else{
-            this.project = tokens.join('/')
-        }
+        this.project = tokens.join('/')
         this.config = config ?: new ProviderConfig('azurerepos')
         this.continuationToken = null
     }
+    /**
+     An Azure repo is identified with the Organization/Project/Repository parameters.
+     This function gets these parameters for the different URL path formats supported in Nextflow for Azure repositories.
+
+     @param urlPath Path of the Azure repo URL
+     @return List with the azure repo parameters with the following order [ Organization, Project, Repository ]
+     **/
+    static List<String> getUniformPath(String urlPath){
```

Comment:
at cost to be repetitive let's add a unit test for this method testing all branches

Reply:
Added in the last commit

Reply:
Nice thanks